pkg/webhook/controlplane/validator.go

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlplane
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Validator validates and if needed mutates objects.
+type Validator interface {
+	// Validate validates and if needed mutates the given object.
+	Validate(ctx context.Context, obj runtime.Object) error
```

Comment:
From a naming perspective, it'd be nice to align with the k8s [admission plugin interface](https://github.com/kubernetes/apiserver/blob/87b5ac0c063e0b28c5a07f664e260268fd1fa3e5/pkg/admission/interfaces.go#L101).

WDYT?

Reply:
I think the `Validator` interface is different and we should choose a name accordingly. It doesn't admit a request, instead it validates the given object for meeting certain conditions,  and if it doesn't, mutates it accordingly. I hesitated between `Validator.Validate`, `Mutator.Mutate`, and `Ensurer.Ensure`. I chose `Validator` just because I thought it was the most conventional of the three. What would your preference be? @rfranzke, any thoughts?

Reply:
Okay got it.

I'd personally vote for `Mutator.Mutate` since it somehow feels like an abuse if mutation happens in a validation function.

Reply:
Tend to agree, renamed.